### PR TITLE
Add machine time calculation functions for DL_CCVM

### PR DESCRIPTION
**Pull request by KatieO4t1qbit**
This commit adds an option to calculate the machine_time using DL_CCVM. This option can only be used if the solver is DLSolver.

Closes https://github.com/1QB-Information-Technologies/ccvm/issues/158

Future commits (other PRs) will add the calculations for other optics solvers.

Testing
You can verify this feature by:

* running the new unit tests
* customizing the solver type and/or the machine parameter of machine_time() in the examples/ccvm_boxqp_plot.py file and then running it

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [ ] The correct base branch is being used, if not `main`

### CODE REVIEW

File: ccvm_simulators/tests/unit/solvers/test_ccvm_solver.py [hunk 1]
```diff
@@ -347,3 +352,31 @@ def test_machine_time_dl_solver_with_gpu_machine(self):
         self.assertEqual(
             gpu_callable(dataframe=dataframe, problem_size=problem_size), 30.0
         )
+
+    def test_machine_time_dl_solver_with_dl_machine(self):
```
Comment (Estella-Wang-1qbit) [inline ccvm_simulators/tests/unit/solvers/test_ccvm_solver.py:356]: Nice test!

**Review by Estella-Wang-1qbit — APPROVED** [review]
Thanks Katie. Looks good 👍

File: ccvm_simulators/solvers/dl_solver.py [hunk 2]
```diff
@@ -357,6 +357,59 @@ def _optics_machine_energy_callable(matching_df: DataFrame, problem_size: int):
...
+                ValueError: when the given dataframe does not contain the required
+                    columns.
+
+            Returns:
+                float: The average average time spent by the solver on a single instance.
+            """
+            self._validate_machine_time_dataframe_columns(dataframe)
+
+            laser_clock = machine_parameters["laser_clock"]
+            # TODO before merge: Ask Farhad if using mean was correct here
```
Comment (ChanWooY-1QBit) [inline ccvm_simulators/solvers/dl_solver.py:402]: Make sure to ask this and remove it before merging.

**Review by ChanWooY-1QBit — APPROVED** [review]
Looks great! Just make sure to address my comment before merging.

**Pull request merged**